lib/matplotlib/gridspec.py

```diff
@@ -484,7 +484,12 @@ def __init__(self, nrows, ncols,
         """
         self._wspace = wspace
         self._hspace = hspace
-        self._subplot_spec = subplot_spec
+        if isinstance(subplot_spec, SubplotSpec):
```

Comment:
Any reason not to use `_api.check_isinstance`?

Reply:
I didn't realize that existed.  However, that issues a generic type error, whereas we want to provide a bit of guidance here?